### ISSUE TITLE
Device column not getting populated correctly while running the same feature tests on different devices and browsers

### ISSUE THREAD

**Issue by violetastegaru** [damianszczepanik/cucumber-reporting#494]
Using the Jenkins Plugin: Cucumber-jvm reports 2.5.1 to consolidate the cucumber reports that are coming from different platform configurations under test.

For example:
- several feature tests are triggered by a main batch to run in parallel on 2 different browsers.
- Login and Logout Jenkins Jobs is a multi-configuration job that triggers the WebDriver execution against chrome and firefox in the same time. These combinations might include mobile devices as well.
The json reports will be stored under: loginlogout/chrome_cucumber.json and firefox_cucumber.json

As a pattern, there will be n feature tests executed in parallel on these browsers generating the cucumber json reports that will be aggregated using the Jenkins Cucumber-jvm reports plugin.

Feature_1/chrome_cucumber.json
Feature_1/firefox_cucumber.json
………………….
Feature_n/chrome_cucumber.json
Feature_n/firefox_cucumber.json

In the end, we would like to aggregate all the Cucumber reports, but we end up in having multiple entries for the features, which is hard to follow if there are several browser and devices used for testing.
I have enabled the parallel testing checkbox in all the Cucumber JVM report publishers, but in the end my problem is that I cannot see the Device column populated.
 
![image](https://cloud.githubusercontent.com/assets/20298849/16582841/e495f176-42b1-11e6-8c2d-9b41a36907fa.png)

Also if I click on a Feature link, I see the Feature Report table kind of broken, the  device colum is only listed as part of the header.

 
![image](https://cloud.githubusercontent.com/assets/20298849/16582843/e9461250-42b1-11e6-984d-cf07ea1f9830.png)

Have i misunderstood how this plugin works, or these are bugs?
Thank you!

**Comment by damianszczepanik** [comment]
Add json file without changing the name of the file. Looks like the bug but it might be resolved recently even not released

**Comment by violetastegaru** [comment]
[aggregate.zip](https://github.com/damianszczepanik/cucumber-reporting/files/348101/aggregate.zip)
I have added some examples in the attached archive. Thank you!

**Comment by damianszczepanik** [comment]
Right, the plugin assumes that the device name is taken from file name so if both files are the same - such case is not handled properly.

What would you do in such case, what should be displayed as device name?

**Comment by violetastegaru** [comment]
In our framework, we take a feature(Registration) and execute the same on different platform combinations, might be desktop, mobile web or might be mobile devices. My initial assumption was that the device name could make the difference when we have multiple entries for the same feature based on configuration. 
Something like:
![image](https://cloud.githubusercontent.com/assets/20298849/16617274/19a7f69e-4383-11e6-9084-acd7cb4e9b17.png)

**Comment by damianszczepanik** [comment]
You are right.
The problem is that the only input report has is a json files that have no such information and the file name which are the same in your case. So having those two how would you evaluate device name?

**Comment by damianszczepanik** [comment]
So device name is taken from the JSON file name. I suggest to rename files so each one is unique. I'd rather want the user to set device name as the file name than developing magic code that renames device name to something different when files are the same.

**Comment by violetastegaru** [comment]
I am following your advice to keep the file names unique. I have tested the latest updates for the Jenkins Plugin | Cucumber-JVM Reports(2.6.2 and 2.6.3) and it work perfectly. Many thanks!
